### PR TITLE
Projectiles Update

### PR DESCRIPTION
**Pull request by KayenEQ**
Fixed bug with spell projectiles causing angle to not be calculated correctly. Spell projectiles will now fire straight instead of from the side.

Significantly improved all projectile timings. At least up to range of 300 should see more accurate timings for damage/effects occurring upon projectile impact.

Will be noticed most significantly for all spell projectiles and for longer range archery.

### CODE REVIEW

**Review by mackal — APPROVED** [review]
Code wise, this looks fine. I trust @KayenEQ that it looks right in game, but I'll hold off on merging for a bit to give others to actually test in game.

**Comment by SecretsOTheP** [comment]
Looks safe to me. I hate parameter spaghetti (should really be a struct) but we use that elsewhere in the attack code still.

**Comment by mackal** [comment]
Okay, I did test this on my server, looks fine, I'll probably merge this evening unless someone else finds issue.

**Pull request merged**
